### DISCUSSION TITLE
Unable to login via UART connection

### DISCUSSION THREAD

**Discussion by dmirecki** [Hypfer/Valetudo#2052] [Q&A / Support] [answered]
### Are you already using Valetudo?

Yes, for a few months

### Your questions

Hi,

I sent my Dreame L10 Pro to the seller for repair (I have a warranty). The seller replaced the motherboard and sent the robot back to me. Factory settings were restored, so I want to install valetudo again, but I can't log in via UART. In console, I see: `p2029_release login: `. I'm using `root` as login and password generated based on the SN, but I am not able to login (text `Login incorrect` appears).

The sticker under the dustbin has a format `PXXXXXXXSEUXXXXX01WO:9847`. I was trying to using `PXXXXXXXSEUXXXXX01`, `PXXXXXXXSEUXXXXX01WO`, `PXXXXXXXSEUXXXXX01WO9847`, `PXXXXXXXSEUXXXXX01WO:9847` as the SN in password generator, but still the password appears to be incorrect.

Is it possible that the SN was changed because of new motherboard?

**Comment by stefanmd023** [comment]
STOP right now.

replaced motherboards are shipping with a new firmware that gets bricked if you try to root it with the current instructions. The devs will be looking into it but there is no guarantee it will be resolved.

**Reply by dmirecki** [reply]
Thanks for your answer! 

So, the problem with login is a symptom that the new firmware was used, right? Is there any ticket or thread where I can check if there is any progress with solving the issue?

**Reply by stefanmd023** [reply, marked as answer]
problem with login always happens when they change the motherboard, because the serial number on the outside used to calculate the password does not match anymore, irrespective of the firmware.

But since a certain date, replaced motherboards are being released with a firmware that bricks with the current rooting process. THis has happened to people who insisted on finding the password despite warnings and ended up bricking their robot.

This firmware has not been released to the public yet and no there is no ticket tracking the progress because it is not an issue.

**Reply by hannesdonel** [reply]
@stefanmd023 where do you have that information from? I just can't find anything on the internet.

I'm trying to root my Xiaomi X10+ that was bought in April directly from Xiaomi store and experience the same. But I can't relate that 100% to firmware. What makes me curious is:
- serial has a very strange format. I send a picture of it to Xiaomi and the confirmed it to be correct. Format is: 49717/BZAH2****08681 Does anyone else have something like that? It's an EU model.
- the login promt appears 3 times in a row. I flashed my USB stick with dd and the UART image from valetudo. But it seems to spawn multiple shells. How can I troubleshoot that?

Thanks in advance.

**Reply by stefanmd023** [reply]
@hannesdonel you don't necessarily find everything on the internet. This is happening now and may or may not stop happening. That's possibly the first time it was written black on white on the internet. It is very specific information so you can't just relate it to your newly bought X10+.

Regarding multiple shells, this is usually an issue with how the usb was formatted, especially if it has multiple partitions. Try to use the image which is in the instructions, or try using another usb drive.

**Reply by hannesdonel** [reply]
@stefanmd023 Got you. I was rather looking for similiar behavior in case that manufacturers patch their firmware. My understanding to that point was that if this case happens and they patch their devices it wouldn't be possible to spawn an UART shell from USB device at all.

I tried multiple sticks, formatted them exactly like instructed by valetudo and flashed valetudos dreame UART .img to them. Always the same behavior.

I really have no clue at all how to proceed here. Would be great to bypassing buying multiple device just to see if they behave the same :-D
